src/helpers/Bmp.cpp

```diff
@@ -0,0 +1,110 @@
+#include "Bmp.hpp"
+
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+class BmpHeader {
+public:
+    unsigned char format[2];
+    uint32_t sizeOfFile;
+    uint16_t reserved1;
+    uint16_t reserved2;
+    uint32_t dataOffset;
+    uint32_t sizeOfBitmapHeader;
+    uint32_t width;
+    uint32_t height;
+    uint16_t numberOfColors;
+    uint16_t numberOfBitPerPixel;
+    uint32_t compressionMethod;
+    uint32_t imageSize;
+    uint32_t horizontalResolutionPPM;
+    uint32_t verticalResolutionPPM;
+    uint32_t numberOfCollors;
+    uint32_t numberOfImportantCollors;
+
+    BmpHeader(std::ifstream& file) {
+        file.read(reinterpret_cast<char*>(&format), sizeof(format));
+        if (!(format[0] == 66 and format[1] == 77)) {
+            Debug::log(ERR, "Unable to parse bitmap header: wrong bmp file type");
+            exit(1);
+        }
+        file.read(reinterpret_cast<char*>(&sizeOfFile), sizeof(sizeOfFile));
```

Comment:
can this be exploited? what if the file ends prematurely?

Reply:
stream.read() reads until the end of file, so, if we want read more bytes then file have, we get only existing bytes plus eof

Reply:
added a check for equality of the real file size and the size from the header